src/weapon/w_043.c

```diff
@@ -9,17 +9,76 @@ extern s32 g_HandId;
 #define g_Animset w_043_1
 #define g_Animset2 w_043_2
 
-extern SpriteParts D_131000_8017A040[];
 extern AnimationFrame D_131000_8017AF40[];
 
 INCLUDE_ASM("weapon/nonmatchings/w_043", EntityWeaponAttack);
 
-INCLUDE_ASM("weapon/nonmatchings/w_043", func_ptr_80170004);
+s32 func_ptr_80170004(Entity* self) {
+    switch (self->step) {
+    case 0:
+        self->palette = PAL_OVL(0x100);
+        self->zPriority = PLAYER.zPriority;
+        self->facingLeft = PLAYER.facingLeft;
+        // this toggles the high bit, but xor does not
+        // produce the same set of instructions
+        self->animCurFrame = PLAYER.animCurFrame + 0x8000;
```

Comment:
Looks like this can either be an add or subtract (since it wraps around) but I think it's the same constant as the `ANIMSET_OVL_FLAG`: https://decomp.me/scratch/xyrKw

Reply:
0x8000 seem to indicate that a new animation frame needs to be loaded to vram or something similar https://github.com/Xeeynamo/sotn-decomp/blob/899226c3c305cfa78ab878c2f25fb8e5df1e6234/src/dra/4A538.c#L872

Reply:
The `ANIMSET_OVL` should work in this case, I agree

Reply:
I think using the constant sounds good. I tried a variety of different things to generate matching code because it does seem like a toggle, but this is the only way I found to generate the same assembly. The behavior is a toggle, rather than explicitly setting the value like the macro does, even if that's not what was originally intended.